### DISCUSSION TITLE
How can I run an application by sgx simulator mode in gramine(docker image: gramineproject/gramine)?

### DISCUSSION THREAD

**Discussion by fffmonkeyking** [gramineproject/gramine#1559] [General]
How can I run an application by sgx simulator mode in gramine(docker image: gramineproject/gramine)?

**Comment by dimakuv** [comment]
There's not really an *SGX simulator mode* in Gramine.

There is however a mode that you can consider "very close to the simulator" -- it is the `gramine-direct` mode.

To run it, you should simply replace `gramine-sgx` with `gramine-direct`. E.g. `gramine-direct helloworld`.

**Comment by fffmonkeyking** [comment]
another question:
Can I run a pytorch example base on cuda by gramine-direct/gramine-sgx ?

**Reply by dimakuv** [reply]
> Can I run a pytorch example base on cuda by gramine-direct/gramine-sgx ?

No, you can't run any GPU-specific applications/workloads on Gramine yet.

**Reply by fffmonkeyking** [reply]
Do you have plans to support running the GPU specific applications/workloads on Gramine?

If there is the plan, when is the expected availability of this feature?

**Reply by dimakuv** [reply]
> If there is the plan, when is the expected availability of this feature?

This is unfortunately unclear...

**Comment by fffmonkeyking** [comment]
Has the feature described in this document <Computation offloading to hardware accelerators in Intel® SGX and Gramine Library OS> been implemented in Gramine ?
(URL link for document: https://arxiv.org/ftp/arxiv/papers/2203/2203.01813.pdf)

**Reply by dimakuv** [reply]
Not yet (not fully yet). Should be implemented by the end of this year.

**Comment by fffmonkeyking** [comment]
Thank you very much for your quick response
